### PR TITLE
Fixes Piston Bugs (GlobalBlockPalette + sticky piston not pulling other sticky piston)

### PR DESCRIPTION
**Pull request by PocketMiner82**
Exception was thrown:
```
2020-07-07 17:47:21.504 [Application-Thread] ERROR - Welt "world" konnte nicht getickt werden: java.lang.IllegalArgumentException: Empty string not allowed
	at cn.nukkit.nbt.tag.StringTag.<init>(StringTag.java:19)
	at cn.nukkit.nbt.tag.CompoundTag.putString(CompoundTag.java:99)
	at cn.nukkit.block.BlockPistonBase.doMove(BlockPistonBase.java:231)
	at cn.nukkit.block.BlockPistonBase.checkState(BlockPistonBase.java:150)
	at cn.nukkit.block.BlockPistonBase.onUpdate(BlockPistonBase.java:120)
	at cn.nukkit.level.Level.updateAroundRedstone(Level.java:1257)
	at cn.nukkit.block.BlockRedstoneWire.calculateCurrentChanges(BlockRedstoneWire.java:155)
	at cn.nukkit.block.BlockRedstoneWire.updateSurroundingRedstone(BlockRedstoneWire.java:95)
	at cn.nukkit.block.BlockRedstoneWire.onUpdate(BlockRedstoneWire.java:231)
	at cn.nukkit.level.Level.updateAroundRedstone(Level.java:1257)
	at cn.nukkit.block.BlockRedstoneTorch.onBreak(BlockRedstoneTorch.java:90)
	at cn.nukkit.level.Level.useBreakOn(Level.java:2139)
	at cn.nukkit.level.Level.useBreakOn(Level.java:1999)
	at cn.nukkit.level.Level.useBreakOn(Level.java:1994)
	at cn.nukkit.level.Level.useBreakOn(Level.java:1990)
	at cn.nukkit.level.Level.useBreakOn(Level.java:1986)
	at cn.nukkit.level.Level.useBreakOn(Level.java:1982)
	at cn.nukkit.level.Level.useBreakOn(Level.java:1978)
	at cn.nukkit.block.BlockTorch.onUpdate(BlockTorch.java:56)
	at cn.nukkit.block.BlockRedstoneTorch.onUpdate(BlockRedstoneTorch.java:98)
	at cn.nukkit.level.Level.doTick(Level.java:848)
	at cn.nukkit.Server.checkTickUpdates(Server.java:1108)
	at cn.nukkit.Server.tick(Server.java:1190)
	at cn.nukkit.Server.tickProcessor(Server.java:962)
	at cn.nukkit.Server.start(Server.java:930)
	at cn.nukkit.Server.<init>(Server.java:588)
	at cn.nukkit.Nukkit.main(Nukkit.java:120)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at de.dytanic.cloudnet.wrapper.Wrapper.lambda$startApplication$5(Wrapper.java:532)
	at java.lang.Thread.run(Thread.java:748)
```

### CODE REVIEW

**Comment by MrRazamataz** [comment]
This version seems to break enchanting tables again.

**Comment by tuilakhanh** [comment]
> This version seems to break enchanting tables again. 

I dont think soo. It only changes in piston

File: src/main/java/cn/nukkit/block/BlockPistonBase.java [hunk 1]
```diff
@@ -228,7 +228,7 @@ private boolean doMove(boolean extending) {
                                 .putInt("id", newBlock.getId()) //only for nukkit purpose
                                 .putInt("meta", newBlock.getDamage()) //only for nukkit purpose
                                 .putShort("val", newBlock.getDamage())
-                                .putString("name", GlobalBlockPalette.getName(newBlock.getId()))
+                                .putString("name", newBlock.getName())
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/block/BlockPistonBase.java:231]: The client expect it to be a valid string-id like `minecraft:cobblestone`, that information is only accurately available at the `runtime_block_states.data` files which are (currently) interpreted by the `GlobalBlockPalette`. If an empty string got into that spot, than the issue is with the names registered in the `GlobalBlockPalette` class

**Comment by PocketMiner82** [comment]
I tried it with stone and grass block

**Comment by PocketMiner82** [comment]
Found the issue in Global Block Palette, you never set the contents of legacyIdToString Map.

**Comment by MrRazamataz** [comment]
On the latest snapshot I get stuck on loading resources.

**Comment by PocketMiner82** [comment]
I'm currently testing, why a sticky piston can't pull back another sticky piston, when this other sticky piston has a block on the face:
2020-07-08_11h46_40
2020-07-08_11h47_06

**Comment by PocketMiner82** [comment]
Something with BlockEntityPistonArm#movable is wrong...

**Comment by PocketMiner82** [comment]
Sorry, but I don't know how to fix it. Any idea @joserobjr? I'm very sure, that the Error is caused by BlockEntityPistonArm#isMovable()

**Comment by PocketMiner82** [comment]
The error seems to be here: https://github.com/GameModsBR/PowerNukkit/blob/master/src/main/java/cn/nukkit/blockentity/BlockEntityPistonArm.java#L194

**Comment by PocketMiner82** [comment]
Now everything should be fine :D

**Comment by PocketMiner82** [comment]
I made lots of test ingame (I also built a jeb door, lol) and seems to not have bugs, did you find some?

**Comment by joserobjr** [comment]
Ok, since it looks nice and you tested a lot, I'm going to trust your tests and merge into `bleeding` before my own tests :)

Thank you very much!

**Comment by PocketMiner82** [comment]
IMG_20200708_204242
IMG_20200708_204311

Found this bug. I placed the redstone blocks from right to left, the last piston doesn't push out, I'll try to fix it and create a new PR (tomorrow or friday).

**Comment by PocketMiner82** [comment]
BlockEntities of blocks behind the block, directly at the pistons don't get moved, trying to fix it in new pr...

**Pull request merged**